codex/purchasing/states/failed.nim

```diff
@@ -0,0 +1,9 @@
+import ../statemachine
+import ./error
+
+type
+  PurchaseFailed* = ref object of PurchaseState
+
+method enter*(state: PurchaseFailed) =
+  let error = newException(PurchaseError, "Purchase failed")
```

Comment:
It might be helpful to add another layer of information here by describing the error type:
```suggestion
  type PurchaseFailedError* = ref object of of PurchaseError
  let error = newException(PurchaseFailedError, "Purchase failed")
```

Reply:
Seems a bit of overkill to me for now, but if we need it we can add it later.